Gems/AWSCore/Code/Source/Configuration/AWSCoreConfiguration.h

```diff
@@ -29,6 +29,8 @@ namespace AWSCore
         static constexpr const char AWSCoreDefaultProfileName[] = "default";
         static constexpr const char AWSCoreProfileNameKey[] = "/AWSCore/ProfileName";
 
+        static constexpr const char AWSCoreAllowAWSMetadataQueriesKey[] = "/AWSCore/AllowAWSMetadataQueries";
```

Comment:
Minor: Would be curious if theres a general collection of Metadata services or this should be more specific to EC2 IMDS

Reply:
there are at least 2 (EC2 + ECS) that are part of the default provider chain (plus who knows what the future may hold 🔮 ). The latter we don't support yet but is a good candidate for a feature request, IMO.